src/goPlayground.ts

```diff
@@ -0,0 +1,76 @@
+import vscode = require('vscode');
+import { execFile } from 'child_process';
+import { outputChannel } from './goStatus';
+import { getBinPath } from './util';
+import { promptForMissingTool } from './goInstallTools';
+
+// isENOENT checks if the given error results from a missing tool installation
+export const isENOENT = (err: Error): Boolean => (
+	!!err && (<any>err).code === 'ENOENT'
+);
+
+// flags describes the configuration toggles for the command
+type flags = { [key: string]: Boolean };
+
+// IPlaygroundUploader needs to be implemented by the uploader passed to createCommandWith
+export interface IPlaygroundUploader {
+	upload(code: string, config: flags): Promise<string>;
+}
+
+// createCommandWith retrieves the go.playground configuration and passes
+// it to the given `uploader`, together with the current editor selection
+// (or the full content of the editor window if the selection is empty)
+export const createCommandWith = (uploader: IPlaygroundUploader) => (): Promise<any> => {
+	const editor = vscode.window.activeTextEditor;
+	const config: flags = vscode.workspace.getConfiguration('go', editor.document.uri).get('playground');
+
+	const selection = editor.selection;
+	const code = selection.isEmpty
+		? editor.document.getText()
+		: editor.document.getText(selection);
+
+	outputChannel.show();
+	outputChannel.appendLine('Upload to the Go Playground in progress...\n');
+
+	return uploader.upload(code, config)
+		.then(result => outputChannel.append(result))
+		.catch(err =>  {
+			if ((<any>err).missingTool) {
+				promptForMissingTool(err.missingTool);
+			} else {
+				vscode.window.showErrorMessage(err.message);
```

Comment:
Even after installing `goplay`, user will have to reload VSCode for the change to take affect because the location of the binary gets calculated once in the beginning and isnt updated.

Feel free to use `getBinPath` for every call